+ source /root/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start environmental preparation.'
+ message='Start environmental preparation.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start environmental preparation.'
Wed Oct 11 10:38:58 2023 - INFO  - Start environmental preparation.
+ DNF_INSTALL patchutils
+ pkgs=patchutils
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs patchutils --node 1 --tempfile ''
+ tmpfile2='Last metadata expiration check: 0:03:56 ago on Wed Oct 11 10:35:21 2023.
Error: 
 Problem: conflicting requests
  - nothing provides patch needed by patchutils-0.4.2-1.oe2309.riscv64 from OS
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
+ '[' -z '' ']'
+ tmpfile='Last metadata expiration check: 0:03:56 ago on Wed Oct 11 10:35:21 2023.
Error: 
 Problem: conflicting requests
  - nothing provides patch needed by patchutils-0.4.2-1.oe2309.riscv64 from OS
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
+ mkdir patchutils
+ cd patchutils
+ cp ../common/1.csv ../common/1.txt ../common/2.txt ../common/3.txt ../common/t1.html ./
+ diff -Naur 1.txt 2.txt
+ diff -Naur 2.txt 3.txt
+ LOG_INFO 'End of environmental preparation!'
+ message='End of environmental preparation!'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End of environmental preparation!'
Wed Oct 11 10:39:18 2023 - INFO  - End of environmental preparation!
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Wed Oct 11 10:39:18 2023 - INFO  - Start to run test.
+ flipdiff --combine test1.patch test2.patch
oe_test_patchutils_flipdiff_02.sh: line 35: flipdiff: command not found
+ grep '@@ -1,2 +1,4 @@'
+ CHECK_RESULT 1 0 0 'Check flipdiff --combine test1.patch test2.patch  failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check flipdiff --combine test1.patch test2.patch  failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check flipdiff --combine test1.patch test2.patch  failed'
+ LOG_ERROR 'Check flipdiff --combine test1.patch test2.patch  failed'
+ message='Check flipdiff --combine test1.patch test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check flipdiff --combine test1.patch test2.patch  failed'
Wed Oct 11 10:39:19 2023 - ERROR - Check flipdiff --combine test1.patch test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 36'
+ message='oe_test_patchutils_flipdiff_02.sh line 36'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 36'
Wed Oct 11 10:39:19 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 36
+ '[' 0 -eq 1 ']'
+ return 0
+ flipdiff --flip test1.patch test2.patch
oe_test_patchutils_flipdiff_02.sh: line 37: flipdiff: command not found
+ grep '@@ -1,2 +1,4 @@'
+ CHECK_RESULT 1 0 0 'Check flipdiff --flip test1.patch test2.patch  failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check flipdiff --flip test1.patch test2.patch  failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check flipdiff --flip test1.patch test2.patch  failed'
+ LOG_ERROR 'Check flipdiff --flip test1.patch test2.patch  failed'
+ message='Check flipdiff --flip test1.patch test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check flipdiff --flip test1.patch test2.patch  failed'
Wed Oct 11 10:39:19 2023 - ERROR - Check flipdiff --flip test1.patch test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 38'
+ message='oe_test_patchutils_flipdiff_02.sh line 38'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 38'
Wed Oct 11 10:39:20 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 38
+ '[' 0 -eq 1 ']'
+ return 0
+ flipdiff --no-revert-omitted test1.patch test2.patch
oe_test_patchutils_flipdiff_02.sh: line 39: flipdiff: command not found
+ grep '@@ -1,2 +1,4 @@'
+ CHECK_RESULT 1 0 0 'Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
+ LOG_ERROR 'Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
+ message='Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check flipdiff --no-revert-omitted test1.patch test2.patch  failed'
Wed Oct 11 10:39:20 2023 - ERROR - Check flipdiff --no-revert-omitted test1.patch test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 40'
+ message='oe_test_patchutils_flipdiff_02.sh line 40'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 40'
Wed Oct 11 10:39:20 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 40
+ '[' 0 -eq 1 ']'
+ return 0
+ flipdiff --in-place test1.patch test2.patch
oe_test_patchutils_flipdiff_02.sh: line 41: flipdiff: command not found
+ CHECK_RESULT 127 0 0 'Check flipdiff --in-place test1.patch test2.patch  failed'
+ actual_result=127
+ expect_result=0
+ mode=0
+ error_log='Check flipdiff --in-place test1.patch test2.patch  failed'
+ exit_mode=0
+ '[' -z 127 ']'
+ '[' 0 -eq 0 ']'
+ test 127x '!=' 0x
+ test -n 'Check flipdiff --in-place test1.patch test2.patch  failed'
+ LOG_ERROR 'Check flipdiff --in-place test1.patch test2.patch  failed'
+ message='Check flipdiff --in-place test1.patch test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check flipdiff --in-place test1.patch test2.patch  failed'
Wed Oct 11 10:39:21 2023 - ERROR - Check flipdiff --in-place test1.patch test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 42'
+ message='oe_test_patchutils_flipdiff_02.sh line 42'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 42'
Wed Oct 11 10:39:21 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 42
+ '[' 0 -eq 1 ']'
+ return 0
+ filterdiff --grep ... test2.patch
oe_test_patchutils_flipdiff_02.sh: line 43: filterdiff: command not found
+ grep 2.txt
+ CHECK_RESULT 1 0 0 'Check filterdiff --grep ... test2.patch  failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check filterdiff --grep ... test2.patch  failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check filterdiff --grep ... test2.patch  failed'
+ LOG_ERROR 'Check filterdiff --grep ... test2.patch  failed'
+ message='Check filterdiff --grep ... test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check filterdiff --grep ... test2.patch  failed'
Wed Oct 11 10:39:22 2023 - ERROR - Check filterdiff --grep ... test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 44'
+ message='oe_test_patchutils_flipdiff_02.sh line 44'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 44'
Wed Oct 11 10:39:22 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 44
+ '[' 0 -eq 1 ']'
+ return 0
+ filterdiff -X pattren --lines=-5 test2.patch
oe_test_patchutils_flipdiff_02.sh: line 45: filterdiff: command not found
+ grep '@@ -1,2 +1,4 @@'
+ CHECK_RESULT 1 0 0 'Check filterdiff -X pattren --lines=-5 test2.patch  failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='Check filterdiff -X pattren --lines=-5 test2.patch  failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'Check filterdiff -X pattren --lines=-5 test2.patch  failed'
+ LOG_ERROR 'Check filterdiff -X pattren --lines=-5 test2.patch  failed'
+ message='Check filterdiff -X pattren --lines=-5 test2.patch  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check filterdiff -X pattren --lines=-5 test2.patch  failed'
Wed Oct 11 10:39:22 2023 - ERROR - Check filterdiff -X pattren --lines=-5 test2.patch  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 46'
+ message='oe_test_patchutils_flipdiff_02.sh line 46'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 46'
Wed Oct 11 10:39:23 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 46
+ '[' 0 -eq 1 ']'
+ return 0
+ fixcvsdiff -p -b 1.csv
oe_test_patchutils_flipdiff_02.sh: line 47: fixcvsdiff: command not found
+ CHECK_RESULT 127 0 0 'Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
+ actual_result=127
+ expect_result=0
+ mode=0
+ error_log='Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
+ exit_mode=0
+ '[' -z 127 ']'
+ '[' 0 -eq 0 ']'
+ test 127x '!=' 0x
+ test -n 'Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
+ LOG_ERROR 'Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
+ message='Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed'
Wed Oct 11 10:39:23 2023 - ERROR - Check fixcvsdiff -p -b --deep-brainwave-mode --recurse --compare 1.csv  failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_patchutils_flipdiff_02.sh line 48'
+ message='oe_test_patchutils_flipdiff_02.sh line 48'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_patchutils_flipdiff_02.sh line 48'
Wed Oct 11 10:39:23 2023 - ERROR - oe_test_patchutils_flipdiff_02.sh line 48
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Wed Oct 11 10:39:24 2023 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z 7
+ test 7 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Wed Oct 11 10:39:24 2023 - ERROR - failed to execute the case.
+ exit 7
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Wed Oct 11 10:39:24 2023 - INFO  - Start to restore the test environment.
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z Last metadata expiration check: 0:03:56 ago on Wed Oct 11 10:35:21 2023.
Error: 
 Problem: conflicting requests
  - nothing provides patch needed by patchutils-0.4.2-1.oe2309.riscv64 from OS
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages) ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile 'Last metadata expiration check: 0:03:56 ago on Wed Oct 11 10:35:21 2023.
Error: 
 Problem: conflicting requests
  - nothing provides patch needed by patchutils-0.4.2-1.oe2309.riscv64 from OS
(try to add '\''--skip-broken'\'' to skip uninstallable packages or '\''--nobest'\'' to use not only best candidate packages)'
Traceback (most recent call last):
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 192, in <module>
    exitcode = rpm_remove(args.node, args.pkgs, args.tempfile)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 163, in rpm_remove
    with open(tmpfile, "r") as f:
         ^^^^^^^^^^^^^^^^^^
OSError: [Errno 36] File name too long: "Last metadata expiration check: 0:03:56 ago on Wed Oct 11 10:35:21 2023.\nError: \n Problem: conflicting requests\n  - nothing provides patch needed by patchutils-0.4.2-1.oe2309.riscv64 from OS\n(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)"
+ '[' 0 -ne 0 ']'
+ cd ..
+ rm -rf patchutils
+ LOG_INFO 'Finish restore the test environment.'
+ message='Finish restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Finish restore the test environment.'
Wed Oct 11 10:39:26 2023 - INFO  - Finish restore the test environment.
